cmd/frontend/graphqlbackend/cody_keyword_context.go

```diff
@@ -0,0 +1,265 @@
+package graphqlbackend
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	"github.com/bbalet/stopwords"
+	"math"
+	"regexp"
+	"strings"
+
+	stemmer "github.com/reiver/go-porterstemmer"
+)
+
+var keywordSearchPatternType = "regexp"
+
+const MAX_FILES_TO_RANK = 200
+const KEYWORD_CONTEXT_SEARCH_TIMEOUT = "0.5s"
```

Comment:
Two nits:

First, standard go style is to still use `camelCase` for consts. Capitalizing these exports them, which makes things like `unused` lints ineffective.

Second, we can use `time.Duration` instead of a string constant to keep this well-typed.

```suggestion
const maxFilesToRank = 200
const keywordContextSearchTimeout = 500 * time.Millisecond
```